libr/asm/p/asm_mips_gnu.c

```diff
@@ -58,6 +59,29 @@ static int disassemble(struct r_asm_t *a, struct r_asm_op_t *op, const ut8 *buf,
 	disasm_obj.fprintf_func = &generic_fprintf_func;
 	disasm_obj.stream = stdout;
 
+	if(a->cpu && a->cpu != pre_cpu) {
+		if (r_str_casecmp (a->cpu, "mips64r2") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa64r2; 
+		} else if (r_str_casecmp (a->cpu, "mips32r2") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa32r2; 
+		} else if (r_str_casecmp (a->cpu, "mips64") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa64; 
+		} else if (r_str_casecmp (a->cpu, "mips32") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa32; 
+		}
+
+		if (a->features) {
+			disasm_obj.disassembler_options = r_str_new("");
+			if(strstr (a->features, "n64")) {
```

Comment:
Should I use r_str_xxx instead of strstr? But I did not find a suitable API

Reply:
no, this is fine, but you must probably check if a->features != a->prev_features instead of strstr everytime in the case features is not the same, so you save some strstr

Reply:
okay